nft-provider-aggregator/contracts/NFTProviderAggregator.cdc

```diff
@@ -30,154 +31,170 @@ import NonFungibleToken from "./NonFungibleToken.cdc"
 /// resources inoperable, they should be destroyed too.
 ///
-pub contract NFTProviderAggregator {
+access(all) contract NFTProviderAggregator {
 
     /// Event emitted when an Aggregator resource is created
-    pub event AggregatorResourceInitialized(nftTypeIdentifier: String)
-    
+    access(all) event AggregatorResourceInitialized(nftTypeIdentifier: String)
+
     /// Event emitted when a Supplier resource is created
-    pub event SupplierResourceInitialized(
+    access(all) event SupplierResourceInitialized(
         nftTypeIdentifier: String,
         aggregatorUUID: UInt64,
         aggregatorAddressAtCreation: Address?
         )
-    
+
     /// Event emitted when a NFT Provider Capability is added
-    pub event NFTProviderCapabilityAdded(
+    access(all) event NFTWithdrawCapabilityAdded(
         nftTypeIdentifier: String,
         collectionUUID: UInt64,
         collectionAddressAtInsertion: Address
         )
-    
+
     /// Event emitted when a NFT Provider Capability is removed
-    pub event NFTProviderCapabilityRemoved(
+    access(all) event NFTWithdrawCapabilityRemoved(
         nftTypeIdentifier: String,
         collectionUUID: UInt64
         )
 
     /// Storage paths for Aggregator and Supplier resources
-    pub let AggregatorStoragePath: StoragePath
-    pub let SupplierStoragePath: StoragePath
-    
+    access(all) let AggregatorStoragePath: StoragePath
+    access(all) let SupplierStoragePath: StoragePath
+
     /// Public paths for Aggregator{SupplierPublic} capabilities
-    pub let SupplierPublicPath: PublicPath
+    access(all) let SupplierPublicPath: PublicPath
 
     /// Private paths for Aggregator{SupplierFactory}, Aggregator{SupplierAccess},
     /// and Aggregator{NonFungibleToken.Provider} capabilities
-    pub let SupplierFactoryPrivatePath: PrivatePath
-    pub let SupplierAccessPrivatePath: PrivatePath
-    pub let AggregatedProviderPrivatePath: PrivatePath
-    
+    /// Note: These paths are kept for backward compatibility purposes
+    /// (for use with convertPrivateToStoragePath and getPrivateCapPathFromStoragePath)
+    access(all) let SupplierFactoryPrivatePath: PrivatePath
+    access(all) let SupplierAccessPrivatePath: PrivatePath
+    access(all) let AggregatedProviderPrivatePath: PrivatePath
+
     /// Interface that a supplier would commonly use for publicly exposing
     /// the Supplier resource's getter functions through a public capability
     ///
-    pub resource interface SupplierPublic {
-        pub fun getAggregatorUUID(): UInt64
-        pub fun getSupplierAddedCollectionUUIDs(): [UInt64]
-        pub fun getCollectionUUIDs(): [UInt64]
-        pub fun getIDs(): [UInt64]
+    access(all) resource interface SupplierPublic {
+        access(all) view fun getAggregatorUUID(): UInt64
+        access(all) view fun getSupplierAddedCollectionUUIDs(): [UInt64]
+        access(all) view fun getCollectionUUIDs(): [UInt64]
+        access(all) fun getIDs(): [UInt64]
```

Comment:
the `appendAll` built-in method is somehow considered impure operation, agreed it would make more sense but I imagine there must be a reason for that related to the current Cadence language design